### PR TITLE
[2.0] Remove PresetBuildConfiguration in favour of ConfigurationName 

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
As suggested by @kwridan I'm removing `PresetConfiguration` in favor of `ConfigurationName`. I took the opportunity to revisit the interfaces of all the scheme action models from `ProjectDescription` to name the configuration argument consistently, `configuration`. Moreover, I made all the constructors from the actions internal and exposed static methods to have the flexibility to modify the main constructor in the future without breaking the public contract.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/ProjectDescription/ConfigurationName.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+/// It represent's a project configuration.
+public struct ConfigurationName: ExpressibleByStringLiteral, Codable, Equatable {
```
Comment (luispadron) [inline Sources/ProjectDescription/ConfigurationName.swift:4]: Nice!

**Review by luispadron — APPROVED** [review]

**Review by kwridan — APPROVED** [review]
Thanks @pepibumur !

Pushed a few updates to help out with the fixtures

**Comment by kwridan** [comment]
I spotted a few candidates that could be migrated to leverage `ConfigurationName`:
- `CustomConfiguration.name` 
- `Cache.configuration`

**Comment by pepibumur** [comment]
Thanks for the review @kwridan. I updated `Cache.configuration` from `String` to `ConfigurationName`, and I'll leave `CustomConfiguration.name` for another PR where I'll simplify all the build configuration models.

File: Sources/ProjectDescription/RunAction.swift [hunk 2]
```diff
@@ -39,19 +32,20 @@ public struct RunAction: Equatable, Codable {
...
     ///   - arguments: Arguments passed to the process running the app.
     ///   - options: Run action options.
     ///   - diagnosticsOptions: Diagnostics options.
-    public init(config: PresetBuildConfiguration = .debug,
-                executable: TargetReference? = nil,
-                arguments: Arguments? = nil,
-                options: RunActionOptions = .options(),
-                diagnosticsOptions: [SchemeDiagnosticsOption] = [.mainThreadChecker])
+    /// - Returns: Run action.
+    public static func runAction(configuration: ConfigurationName,
```
[the remark this answers is not in the document]
Comment (kwridan) [reply Sources/ProjectDescription/RunAction.swift:41]: I believe the fixture `ios_app_with_custom_scheme` is failing due to method no longer having a default `configuration` - was there a reason it was omitted?

**Review by kwridan — COMMENTED** [review]
Thanks for the updates @pepibumur - can I help with anything to finalise this?

**Comment by pepibumur** [comment]
@kwridan feel free to push the fixes. I haven't found the time to take this to the finish line yet.

**Pull request merged**